meshtastic/tests/seturl/test_cache_manager.py

```
@@ -0,0 +1,252 @@
+        ]
+        mock_local_node.channels = current_channels
+        staged_channel = _make_channel(0, channel_pb2.Channel.Role.PRIMARY, "new")
+
+        def raise_error(msg: str) -> NoReturn:
+            raise ValueError(msg)
+
+        mock_local_node._raise_interface_error = MagicMock(side_effect=raise_error)
+
+        with pytest.raises(
+            ValueError,
+            match="Channel cache changed during replace-all cache update; aborting transaction.",
```

Ruff (0.15.6)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)